src/Gaufrette/Adapter/Local.php

```diff
@@ -309,4 +315,41 @@ protected function createDirectory($directory)
             throw new \RuntimeException(sprintf('The directory \'%s\' could not be created.', $directory));
         }
     }
+
+    /**
+     * Removes files or directories.
+     *
+     * @see https://github.com/symfony/filesystem/blob/master/Filesystem.php#L159
+     *
+     * @param string|array|\Traversable $files A filename, an array of files, or a \Traversable instance to remove
+     *
+     * @return bool
+     */
+    private function doDelete($files)
```

Comment:
It has been implemented already [here](https://github.com/KnpLabs/Gaufrette/blob/master/tests/Gaufrette/Functional/LocalDirectoryDeletor.php), using RecursiveIterator which is cleaner. Could you rather reuse it please?